### PR TITLE
Topic/2117 validate draft

### PR DESCRIPTION
**Pull request by foxlynx**
Lat part of #2117 

# Definition of Done / Review checklist

## Reviewability
- [ ] link to issue
- [ ] note if PR is on top of other PR
- [ ] note if related change in rems-deploy repo
- [ ] consider adding screenshots for ease of review

## API
- [ ] API is documented and shows up in Swagger UI
- [ ] API is backwards compatible or completely new
- [ ] Events are backwards compatible

## Documentation
- [ ] update changelog if necessary
- [ ] add or update docstrings for namespaces and functions
- [ ] components are added to guide page
- [ ] documentation _at least_ for config options (i.e. docs folder)
- [ ] ADR for major architectural decisions or experiments

## Different installations
- [ ] new configuration options added to rems-deploy repository
- [ ] instance specific translations (i.e. LBR kielivara)

## Testing
- [ ] complex logic is unit tested
- [ ] valuable features are integration / browser / acceptance tested automatically

## Accessibility
- [ ] all icons have the aria-label attribute
- [ ] all fields have a label
- [ ] errors are linked to fields with aria-describedby
- [ ] contrast is checked
- [ ] conscious decision about where to move focus after an action

## Follow-up
- [ ] new tasks are created for pending or remaining tasks
- [ ] no critical TODOs left to implement

### CODE REVIEW

File: src/clj/rems/application/commands.clj [hunk 1]
```diff
@@ -341,9 +351,14 @@
                         [event]))))))
 
 (defmethod command-handler :application.command/save-draft
-  [cmd _application _injections]
-  (ok {:event/type :application.event/draft-saved
-       :application/field-values (:field-values cmd)}))
+  [cmd application _injections]
+  (let [forms (:application/forms application)
+        answers (:field-values cmd)
+        answers-formatted (build-index {:keys [:form :field] :value-fn :value} answers)
```
Comment (opqdonut) [inline src/clj/rems/application/commands.clj:357]: you could consider doing `build-index` inside `enrich-form-answers`

File: src/clj/rems/form_validation.clj [hunk 2]
```diff
@@ -26,20 +26,34 @@
...
+  (validate-field-content field))
+
+(defn validate-fields-for-draft [fields]
+  (->> (sort-by :field/id fields)
+       (filter :field/visible)
+       (map validate-draft-field)
+       (remove nil?)
+       (seq)))
+
+(defn validate-fields-for-submit [fields]
```
Comment (opqdonut) [inline src/clj/rems/form_validation.clj:54]: good naming, thanks

File: test/clj/rems/test_form_validation.clj [hunk 3]
```diff
@@ -118,11 +118,11 @@
 
   (testing "error: field input option can be left empty when optional"
     (is (= nil
-           (validate-fields [{:field/id       "1"
-                              :field/title    {:en "Option list."
+           (validate-fields-for-submit [{:field/id "1"
+                              :field/title         {:en "Option list."
                                                :fi "Valintalista."}
```
Comment (opqdonut) [inline test/clj/rems/test_form_validation.clj:123]: indentation looks funny here:
- `:fi` should be below `:en`
- we've typically not aligned map values with each other

Comment (foxlynx) [reply test/clj/rems/test_form_validation.clj]: Fixed

File: test/clj/rems/test_form_validation.clj [hunk 4]
```diff
@@ -1,41 +1,41 @@
 (ns rems.test-form-validation
   (:refer-clojure :exclude [validate-fields])
   (:require [clojure.test :refer :all]
-            [rems.form-validation :refer [validate-fields]]))
+            [rems.form-validation :refer [validate-fields-for-submit validate-fields-for-draft]]))
 
-(deftest test-validate-fields
+(deftest test-validate-fields-for-submit
```
Comment (opqdonut) [inline test/clj/rems/test_form_validation.clj:6]: more indentation errors in this function

File: test/clj/rems/test_form_validation.clj [hunk 5]
```diff
@@ -72,22 +72,22 @@
...
                               :field/visible true
                               :field/value "abcdef"}]))))
 
   (testing "error: field input selected option is invalid"
     (is (= [{:type :t.form.validation/invalid-value :field-id "1"}]
-           (validate-fields [{:field/id       "1"
-                              :field/title    {:en "Option list."
+           (validate-fields-for-submit [{:field/id "1"
+                              :field/title         {:en "Option list."
                                                :fi "Valintalista."}
```
Comment (opqdonut) [inline test/clj/rems/test_form_validation.clj:88]: e.g. here

File: src/clj/rems/application/commands.clj (hunk 1 above)
Comment (foxlynx) [reply src/clj/rems/application/commands.clj]: Done!

File: test/clj/rems/test_form_validation.clj (hunk 5 above)
Comment (foxlynx) [reply test/clj/rems/test_form_validation.clj]: Formatted the whole namespace now.

**Pull request merged**